lsl_autobids/processing_new_files.py

```diff
@@ -1,154 +1,123 @@
-                        warnings.warn("Operation aborted. File will not be overwritten. Please check the file again!", UserWarning)
-                        sys.exit()
-                    elif user_response.lower() == 'y':
-                        print('Operation resumed. File will be overwritten with the new file....')
-                        break
-                    else:
-                        print('Invalid response. Please enter "y" for yes or "n" for no.')
-
-                # Skip adding the file with the _old to processed_files if the user chooses to overwrite
-                if user_response.lower() == 'y':
-                    continue  
-            processed_files.append(file_name_without_ext + ext)
-    print("The processed files are: \n")
-    print(processed_files)
+            file_name_no_ext, ext = os.path.splitext(file_name)
+            if file_name_no_ext.endswith('_old'):
+                user_choice = get_user_input("WARNING: The file appears to be a duplicate ('_old'). Overwrite?")
+                if user_choice == 'n':
+                    logger.warning("User declined to overwrite file. Exiting.")
+                    raise RuntimeError("File overwrite declined by user.")
+                logger.info("Overwrite confirmed.")
+                continue 
+            processed_files.append(file_name_no_ext + ext)
+    
+    logger.info(f"Processed files: {processed_files}")
 
-    project_path = os.path.join(project_root,project_name)
     # User prompt asking if we want to proceed to convert and upload
-    ask_convert_message = "Do you want to proceed for BIDS Conversion? (y/n): "
-    while True:
-        user_response = input(ask_convert_message)
-        
-        if user_response.lower() == 'n':
-            warnings.warn("Operation aborted. Files will not be converted.", UserWarning)
-            file = os.path.join(project_path,"last_run_log.txt")
-            with open(file, 'w') as file:
-                file.truncate(0)
-            sys.exit()
-        elif user_response.lower() == 'y':
-            print('Operation resumed. Files will be converted to BIDS.')
-            break
-        else:
-            print('Invalid response. Please enter "y" for yes or "n" for no.')
-   
-    bids_process_and_upload(processed_files, project_name)
+    if cli_args.yes:
+        # Automatically proceed without asking
+        logger.info("Automatically proceeding with BIDS conversion.")
+        logger.info("Starting BIDS conversion.")
+        bids_process_and_upload(processed_files)
+    else:
+        user_choice = get_user_input("Do you want to proceed with BIDS conversion?")
+        if user_choice == 'n':
+            logger.warning("User aborted BIDS conversion.")
+            _clear_last_run_log(project_name)
+            raise RuntimeError("BIDS conversion aborted by user.")
 
-    
+        logger.info("Starting BIDS conversion.")
+        bids_process_and_upload(processed_files)
```

Comment:
I think the logger info and the function call could be moved downwards outside the if/else and then only exist once